### PR TITLE
Json form schema

### PR DESCRIPTION
**Pull request by BhagyashreeBhoyar**

### CODE REVIEW

File: src/assets/css/dataGrid.css [hunk 1]
```diff
@@ -20,16 +20,63 @@ html, body {
...
+@media (max-width: 640px){
+  .field-style {
+      width: 100%;
+      max-width: 100%;
+  }
+  .marks-style {
+      width: 100%;
+      max-width: 100%;
+  }
+    .button-style {
```
Comment (Nikmahajan) [inline src/assets/css/dataGrid.css]: Please follow Indentation.

File: package.json [hunk 2]
```diff
@@ -14,10 +14,11 @@
     "prop-types": "^15.6.2",
     "react": "^16.7.0",
     "react-barcode": "^1.3.4",
+    "react-bootstrap": "^1.0.0-beta.8",
```
Comment (Nikmahajan) [inline package.json]: Why we are using this library.

File: src/assets/css/dataGrid.css [hunk 3]
```diff
@@ -18,6 +18,106 @@ html, body {
...
+        display: block;
+        width: 100%;
+   }
+    .rjsf {
+        padding: 5px 4px;
+        margin: 15px auto;
+    }
+}
+
```
Comment (Nikmahajan) [inline src/assets/css/dataGrid.css]: same here.

File: src/config/memberRegistrationFormSchema.json [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    "classAttended2019"
+  ],
+      "properties": {
+        "name": {
+      "title": "नाम",
+      "type": "string",
+          "customValidator": "meraValidator"
+
+    },
+        "fatherName": {
```
Comment (Nikmahajan) [inline src/config/memberRegistrationFormSchema.json]: Indentations.

File: src/assets/css/dataGrid.css (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/assets/css/dataGrid.css]: Done.

File: src/assets/css/dataGrid.css (hunk 3 above)
Comment (BhagyashreeBhoyar) [reply src/assets/css/dataGrid.css]: Done.

File: src/components/routeComponents/MemberRegistrationCorrectionForm.jsx [hunk 5]
```diff
@@ -0,0 +1,538 @@
...
+        <Popup>
+          <h5>{infoUpdateSuccessMessage}</h5>
+          <LinkButton
+            buttonText={goBackBtnText}
+            linkPath={this.props.context.previousLocation}
+            onClick={() => { this.props.isUpdatedResetAction(); }}
+          />
+        </Popup>
+      );
+    } else if (this.state.isSubmitTriggered && !this.state.isFormChanged && this.state.hasError) {
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/routeComponents/MemberRegistrationCorrectionForm.jsx]: Done.

File: src/config/memberRegistrationFormSchema.json (hunk 4 above)
Comment (BhagyashreeBhoyar) [reply src/config/memberRegistrationFormSchema.json]: Done.

File: src/components/routeComponents/MemberRegistrationForm.jsx [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+  isCreated: false,
+  newStudent: {},
+  setStudentCredentials: () => {},
+};
+
+const mapStateToProps = state => ({
+  newStudent: getNewStudent(state),
+  isCreated: isCreated(state),
+  userType: getUserType(state),
+
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/routeComponents/MemberRegistrationForm.jsx]: Done.

File: src/components/routeComponents/MemberRegistrationCorrectionForm.jsx [hunk 7]
```diff
@@ -0,0 +1,538 @@
...
+  updateStudentData() {
+    const { id, secretKey } = this.props;
+    const { student } = this.state;
+    // Calls api to update student data
+    this.props.updateStudentData({ id,
+      secretKey,
+      student });
+  }
+
+  prePopulateCourse2019(studentData) {
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/routeComponents/MemberRegistrationCorrectionForm.jsx]: Done.

File: src/components/routeComponents/MemberRegistrationCorrectionForm.jsx [hunk 8]
```diff
@@ -0,0 +1,538 @@
...
+import Form from 'react-jsonschema-form';
+import { connect } from 'react-redux';
+import isEmpty from 'lodash/isEmpty';
+import isEqual from 'lodash/isEqual';
+import PropTypes from 'prop-types';
+
+import {
+  formSubmitBtnText,
+  goBackBtnText,
+  infoUpdateSuccessMessage, invalidIdMessage, noInfoChangeMessage,
```
Comment (Nikmahajan) [inline src/components/routeComponents/MemberRegistrationCorrectionForm.jsx]: Indentation

File: package.json (hunk 2 above)
Comment (BhagyashreeBhoyar) [reply package.json]: I have removed this.

File: src/components/routeComponents/MemberRegistrationCorrectionForm.jsx [hunk 9]
```diff
@@ -0,0 +1,538 @@
...
+      buttonText={formSubmitBtnText}
+      type="submit"
+      formName=""
+      value="Submit"
+      onClick={this.handleSubmit}
+    />
+  );
+
+  renderForm = () => {
+    if ((this.props.pageUser === USER_TYPES.STUDENT_WITH_URL || this.props.pageUser === USER_TYPES.STUDENT)
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/routeComponents/MemberRegistrationCorrectionForm.jsx]: resolved this by a creating separate component.

File: src/components/routeComponents/MemberRegistrationCorrectionForm.jsx [hunk 10]
```diff
@@ -0,0 +1,538 @@
...
+      );
+    }
+    return null;
+  };
+
+  renderSuccessMessage() {
+    // if form data is update and valid and submitted successfully.
+    if (this.props.isUpdated) {
+      return (
+        <Popup>
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/routeComponents/MemberRegistrationCorrectionForm.jsx]: It is already a generic component.

File: src/components/routeComponents/MemberRegistrationCorrectionForm.jsx (hunk 8 above)
Comment (BhagyashreeBhoyar) [reply src/components/routeComponents/MemberRegistrationCorrectionForm.jsx]: Done

File: src/components/CorrectionsForm.jsx [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+          transformErrors={transformErrors}
+        />
+      </div>
+    );
+  }
+  return null;
+
+};
+
+CorrectionsForm.propTypes = {
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/CorrectionsForm.jsx]: Done.

File: src/components/form/FormContainer.jsx [hunk 12]
```diff
@@ -1,9 +1,17 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
-const FormContainer = (props) => (
+const FormContainer = props => (
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/form/FormContainer.jsx]: Done.

File: src/components/CorrectionsForm.jsx [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+    return (
+      <div className="form-container">
+        <div className="form-wrapper" ref={formRef}>
+          {renderSuccessMessage()}
+          <JSONSchemaForm
+            showErrorList={false}
+            noHtml5Validate
+            validate={validate}
+            liveValidate
+            schema={onlyOptin2019.Schema}
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/CorrectionsForm.jsx]: Done.

File: src/assets/css/dataGrid.css [hunk 14]
```diff
@@ -18,6 +18,107 @@ html, body {
     display: flex;
 }
 
+.rjsf {
```
Comment (Nikmahajan) [inline src/assets/css/dataGrid.css]: Please add comment why this type of className is used here?

File: src/components/CorrectionsForm.jsx [hunk 15]
```diff
@@ -0,0 +1,251 @@
...
+  student,
+  onChange,
+  transformErrors,
+  submitStudentDataForOnlyOptInCase,
+  changeIsOnlyOptIn2019,
+  renderBackButton,
+  renderSubmitButtons,
+  formRef }) => {
+
+  if ((pageUser === USER_TYPES.STUDENT_WITH_URL || pageUser === USER_TYPES.STUDENT)
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/CorrectionsForm.jsx]: Done

File: src/assets/css/dataGrid.css (hunk 14 above)
Comment (BhagyashreeBhoyar) [reply src/assets/css/dataGrid.css]: This type of class name mention in json form schema.
For css style to form wrapper purpose used that class name.

File: src/components/CorrectionsForm.jsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (Nikmahajan) [reply src/components/CorrectionsForm.jsx]: Nice! now this look clean and perfect to review.

**Pull request closed without merging**